src/main.ts

```diff
@@ -505,40 +488,23 @@ function makeLongUtils(options: Options, bytes: ReturnType<typeof makeByteUtils>
     `,
   );
 
-  const longToString = conditionalOutput(
-    "longToString",
-    code`
-      function longToString(long: ${Long}) {
-        return long.toString();
-      }
-    `,
-  );
-
-  const longToBigint = conditionalOutput(
-    "longToBigint",
-    code`
-      function longToBigint(long: ${Long}) {
-        return BigInt(long.toString());
-      }
-    `,
-  );
-
   const longToNumber = conditionalOutput(
     "longToNumber",
     code`
-      function longToNumber(long: ${Long}): number {
-        if (long.gt(${bytes.globalThis}.Number.MAX_SAFE_INTEGER)) {
+      function longToNumber(int64: { toString(): string }): number {
+        const num = ${bytes.globalThis}.Number(int64.toString());
```

Comment:
I've verified that this check is reliable in https://github.com/stephenh/ts-proto/pull/1057#discussion_r1630141662.